xmake/modules/detect/sdks/find_vstudio.lua

```diff
@@ -152,13 +152,40 @@ function main()
     ,   ["4.2"]  = "VS42COMNTOOLS"
     }
 
-    -- find vs from environment variables 
+    -- find the single current MSVC/VS from environment variables
     local VCInstallDir = os.getenv("VCInstallDir")
-    local VisualStudioVersion = os.getenv("VisualStudioVersion")
-    if VCInstallDir and VisualStudioVersion then
+    if VCInstallDir and (VCInstallDir ~= "") then
+        local VisualStudioVersion = os.getenv("VisualStudioVersion")
+        if not VisualStudioVersion or (VisualStudioVersion == "") then
+
+            -- heuristic for VisualStudioVersion value (early MSVC/VS versions don't set VisualStudioVersion)
+            local VSInstallDir = os.getenv("VSInstallDir") or ""
+            VisualStudioVersion = string.match(VSInstallDir, '(%d+[.]?%d*)\\?%s*$')
+            if not VisualStudioVersion then VisualStudioVersion = string.match(VCInstallDir, '(%d+[.]?%d*)\\VC\\?%s*$') end
```

Comment:
You can write `VSInstallDir:match'(%d+[.]?%d*)\\?%s*$')` to simplify code.

Reply:
I didn't know that... (I'm new to Lua). I'll make the changes here and re-push it.

Reply:
Updated.